datalad/support/annexrepo.py

```diff
@@ -3327,12 +3327,62 @@ def _save_add(self, files, git=None, git_opts=None):
                     if 'error-messages' in r else None,
                     logger=lgr)
 
-    def _save_post(self, message, status, partial_commit):
+    def _save_post(self, message, status, partial_commit,
+                   amend=False, allow_empty=False):
+
+        if amend and self.is_managed_branch() and \
+                self.format_commit("%B").strip() == "git-annex adjusted branch":
+            # We must not directly amend on an adjusted branch, but fix it
+            # up after the fact. That is if HEAD is a git-annex commit.
+            # Otherwise we still can amend-commit normally.
+            # Note, that this may involve creating an empty commit first.
+            amend = False
+            adjust_amend = True
+        else:
+            adjust_amend = False
+
         # first do standard GitRepo business
         super(AnnexRepo, self)._save_post(
-            message, status, partial_commit)
+            message, status, partial_commit, amend,
+            allow_empty=allow_empty or adjust_amend)
         # then sync potential managed branches
         self.localsync(managed_only=True)
+        if adjust_amend:
+            # We committed in an adjusted branch, but the goal is to amend in
+            # corresponding branch.
+
+            adjusted_branch = self.get_active_branch()
+            corresponding_branch = self.get_corresponding_branch()
+            old_sha = self.get_hexsha(corresponding_branch)
+
+            org_commit_pointer = corresponding_branch + "~1"
+            author_name = self.format_commit("%an", org_commit_pointer)
+            author_email = self.format_commit("%ae", org_commit_pointer)
+            author_date = self.format_commit("%ad", org_commit_pointer)
+            old_parent = self.format_commit("%P", org_commit_pointer)
+            new_env = (self._git_runner.env
+                       if self._git_runner.env else os.environ).copy()
+            # `message` might be empty - we need to take it from the to be
+            # amended commit in that case:
+            msg = message or self.format_commit("%B", org_commit_pointer)
```

Comment:
I would prefer, if we could limit the amount of git calls to collect this type of info. What about this:

```py
author_name, author_email, author_date, old_parent, old_message = \
    self.format_commit("%an%x00%ae%x00%ad%x00%P%x00%B").split('\0')
msg = message or old_message
```

Reply:
Yes, I wondered what to use as separator that is certain to not be included in such fields as author_name. `\0` seems like a reasonable solution. Will do.

Reply:
Hehe, nope. `format_commit` already splits by `\0`.